src/plugins/data/common/search/aggs/utils/get_format_with_aggs.test.ts

```diff
@@ -79,6 +79,16 @@ describe('getFormatWithAggs', () => {
     expect(getFormat).toHaveBeenCalledTimes(1);
   });
 
+  test('returns custom label for range if provided', () => {
+    const mapping = { id: 'range', params: {} };
+    const getFieldFormat = getFormatWithAggs(getFormat);
+    const format = getFieldFormat(mapping);
+
+    expect(format.convert({ gte: 1, lt: 20, label: 'custom' })).toBe('custom');
+    // underlying formatter is not called because custom label can be used directly
+    expect(getFormat).toHaveBeenCalledTimes(0);
+  });
```

Comment:
Could you add a test for handling multiple overlapping ranges?

Reply:
What kind of test are you expecting? The formatter only ever gets a single value passed in. Do you want to cover the behavior of the built-in cache?

Reply:
I think I left this in the wrong place- I was worried about the duplicate label case, but this is the wrong place to test it.